mm2src/coins/siacoin.rs

```diff
@@ -750,10 +791,10 @@ impl MarketCoinOps for SiaCoin {
 
     fn wait_for_htlc_tx_spend(&self, _args: WaitForHTLCTxSpendArgs<'_>) -> TransactionFut { unimplemented!() }
 
-    fn tx_enum_from_bytes(&self, _bytes: &[u8]) -> Result<TransactionEnum, MmError<TxMarshalingErr>> {
-        MmError::err(TxMarshalingErr::NotSupported(
-            "tx_enum_from_bytes is not supported for Sia coin yet.".to_string(),
-        ))
+    fn tx_enum_from_bytes(&self, bytes: &[u8]) -> Result<TransactionEnum, MmError<TxMarshalingErr>> {
+        let tx: V2Transaction =
```

Comment:
`impl TryFrom<Vec<u8>> for SiaTransaction`

Could be used here. However, ideally the `TryFrom<Vec<u8>>` can be replaced with `impl TryFrom<&[u8]>` for better versatility.